syntax = "proto3";
package messages;
option go_package = "./messages";

message Timeserie {
  message Metadata {
    int64 ref_secs = 1;
    int64 ref_nanos = 2;
    uint32 period_millis = 3;
  }
  // Only filled for the first message in the stream
  Metadata metadata = 1;
  repeated uint32 offset_millis = 2;
  repeated uint32 kbd_count = 3;
  repeated uint32 mse_count = 4;
  repeated uint32 btn_count = 5;
}

message TimeserieRepo {
  message Entry {
    string filepath = 1;
    int64 start_secs = 2;
    int64 end_secs = 3;
  }
  repeated Entry entries = 1;
}

